src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -765,78 +766,139 @@ function saturation_adjustment(e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
   end
 end
 
-
```

Comment:
Change line 757 (initial guess for secant iteration in saturation adjustment) to:

` T_2 = air_temperature(e_int, PhasePartition(q_tot, FT(0), q_tot-q_v_sat)) # Assume all ice`

This will give a much better initial estimate (though it is not guaranteed to bracket the zero). It is putting only the saturation excess, computed on the basis of `T_1`, into the ice phase.